apps/site/assets/ts/tnm/components/TransitNearMe.tsx

```diff
@@ -42,33 +45,25 @@ const validateModeFilter = (acc: Mode[], mode: string): Mode[] =>
 export const modesFromQuery = (query: QueryParams): Mode[] =>
   query.filter ? query.filter.split(",").reduce(validateModeFilter, []) : [];
 
-export const fetchData = (
-  query: QueryParams,
+export const fetchRealtimeSchedules = async (
+  stops: Stop[],
   dispatch: Dispatch
-): Promise<void> => {
-  if (
-    (query.latitude || query["location[latitude]"]) &&
-    (query.longitude || query["location[longitude]"]) &&
-    window.fetch
-  ) {
-    return window
-      .fetch(
-        `/transit-near-me/api${paramsToString(
-          query,
-          window.encodeURIComponent
-        )}`
-      )
-      .then((response: Response) => {
-        if (response.ok) return response.json();
-        throw new Error(response.statusText);
-      })
-      .then((routes: RouteWithStopsWithDirections[]) =>
-        dispatch(routeSidebarDataAction(routes))
-      )
-      .catch(() => {});
-  }
+): Promise<undefined> => {
+  if (!window.fetch) return;
 
-  return new Promise(resolve => resolve());
+  for (const stop of stops) {
```

Comment:
Did we do any testing on how long this takes to do network requests serially instead of either getting all the data in one network call, or doing an async mapping with a Promise.all resolution?

I suspect the amount of data might mean it's an imperceivable difference but it would be good to know since it's the problem we're solving and offloading some work to the FE.